uisdk/src/main/java/com/karhoo/uisdk/screen/rideplanning/RidePlanningPresenter.kt

```diff
@@ -0,0 +1,298 @@
+
+            val initialLocation = extras.getParcelable<Location>(BookingActivity.Builder.EXTRA_INITIAL_LOCATION)
+            initialLocation?.let {
+                view.setMapLocation(LatLng(it.latitude, it.longitude))
+            }
+
+            RidePlanningStorage.journeyInfo = extras.getParcelable(BookingActivity.Builder.EXTRA_JOURNEY_INFO)
+            RidePlanningStorage.passengerDetails = extras.getParcelable(BookingActivity.Builder.EXTRA_PASSENGER_DETAILS)
+            RidePlanningStorage.bookingComments = extras.getString(BookingActivity.Builder.EXTRA_COMMENTS)
+            RidePlanningStorage.loyaltyInfo = extras.getParcelable(BookingActivity.Builder.EXTRA_LOYALTY_INFO)
+
+            val injectedBookingMetadata = extras.getSerializable(BookingActivity.Builder.EXTRA_META) as? HashMap<String, String>
+            injectedBookingMetadata?.let {
+                RidePlanningStorage.bookingMetadata?.putAll(it)
+            }
+        }
+    }
+
+    override fun hasLocationPermissions(context: Context): Boolean {
+        return hasLocationPermission(context)
+    }
+
+    override fun requestLocationPermissions(context: Context) {
+        ActivityCompat.requestPermissions(
+            context as Activity,
+            arrayOf(
+                Manifest.permission.ACCESS_FINE_LOCATION,
+                Manifest.permission.ACCESS_COARSE_LOCATION
+            ),
+            MY_PERMISSIONS_REQUEST_LOCATION
+        )
+    }
+
+    override fun onReceivedPermissionsResult(
+        context: Context,
+        requestCode: Int,
+        permissions: Array<String>,
+        grantResults: IntArray
+    ): Boolean {
+        when (requestCode) {
+            MY_PERMISSIONS_REQUEST_LOCATION -> {
+                if ((grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED)) {
+                   return true
+                } else {
+                    if (!ActivityCompat.shouldShowRequestPermissionRationale(context as Activity, Manifest.permission.ACCESS_FINE_LOCATION)) {
+                        return false
+                    }
+                }
+                return false
+            }
+        }
+
+        return false;
+    }
+
+    override fun initializeListeners() {
+        RidePlanningStorage.journeyDetailsStateViewModel.viewActions().observe(view.getViewModelStoreOwner() as LifecycleOwner, bindToAddressBarOutputs())
+        RidePlanningStorage.bookingRequestStateViewModel.viewActions().observe(view.getViewModelStoreOwner() as LifecycleOwner, view.checkoutObserver)
+    }
+
+    override fun parseDataFromActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        when {
+            resultCode == Activity.RESULT_OK && requestCode == REQ_CODE_BOOKING_REQUEST_ACTIVITY -> {
+                if (data?.hasExtra(CheckoutActivity.BOOKING_CHECKOUT_PREBOOK_SKIP_RIDE_DETAILS_KEY) == true) {
```

Comment:
This a pretty long method body. Would it be ok with you to extract at least this if else block to another method?

Reply:
Added a TODO for refactoring this once we adopt the RidePlanning